### DISCUSSION TITLE
useContractRead for wagmi v 1.0.9

### DISCUSSION THREAD

**Discussion by gao-nuxos** [wevm/wagmi#2500] [Q&A]
i have a problem when try to useContractRead for check data i'm not sure mistake where point
```
const contractConfig = {
    address: process.env.NEXT_PUBLIC_CONTRACT_ADDRESS,
    abi: commeABI,
}

```
 in component call
 ``` 
 useContractRead({
        ...contractConfig,
        functionName: 'supplyAvailable',
        watch: true,
        onSuccess(data) {
           console.log(data)
        }
    })
    ```
    
 package.json
 
 "dependencies": {
    "@chakra-ui/icons": "^2.0.19",
    "@chakra-ui/react": "^2.6.1",
    "@emotion/react": "^11.11.0",
    "@emotion/styled": "^11.11.0",
    "connectkit": "^1.4.0",
    "eslint": "8.41.0",
    "eslint-config-next": "13.4.4",
    "ethers": "5",
    "framer-motion": "^10.12.16",
    "next": "13.4.4",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "viem": "^0.3.37",
    "wagmi": "^1.0.9"
  }
  in  commeABI
  
  [
  "constructor()",
   "function supplyAvailable() view returns (uint256[])",
  ]

 TypeError: Cannot use 'in' operator to search for 'name' in constructor() 
![image](https://github.com/wagmi-dev/wagmi/assets/83744818/f381c0dc-c1de-4b1c-a568-ae14c96d60e9)

**Comment by tmm** [comment]
Hard to tell what the exact issue is because this isn't a complete reproduction, but looks like your ABI is in the wrong format. You either need to make it a JSON ABI or use the [`parseAbi`](https://viem.sh/docs/abi/parseAbi.html) utility.

**Reply by gao-nuxos** [reply]
Ok it work!! ,  i try  parseAbi(commeABI) Thanks a lot .
